fix golint and make golint fail to circleci

# Description: 

- [ ] Fix current golint issues we have. See log here

- [ ]  Make golint fail in circleci if there an offences ( at moment `make check` will always succeed)

```make check
>>> Reformatting code
pkg/apis/kubic/v1beta1/register.go
/home/dmaiocchi/go/src/github.com/kubic-project/registries-operator/cmd/registries-operator/main.go:1:1: package comment should not have leading space
/home/dmaiocchi/go/src/github.com/kubic-project/registries-operator/cmd/registries-operator/main.go:42:1: comment on exported var Version should be of the form "Version ..."
/home/dmaiocchi/go/src/github.com/kubic-project/registries-operator/cmd/registries-operator/main.go:44:5: exported var Build should have comment or be unexported
/home/dmaiocchi/go/src/github.com/kubic-project/registries-operator/pkg/apis/kubic/v1beta1/registry_types.go:49:1: comment on exported type RegistryCertificateStatus should be of the form "RegistryCertificateStatus ..." (with optional leading article)
/home/dmaiocchi/go/src/github.com/kubic-project/registries-operator/pkg/apis/kubic/v1beta1/registry_types.go:76:1: comment on exported method Registry.GetCertificateSecret should be of the form "GetCertificateSecret ..."
/home/dmaiocchi/go/src/github.com/kubic-project/registries-operator/pkg/apis/kubic/v1beta1/registry_types.go:91:1: receiver name r should be consistent with previous receiver name registry for Registry
/home/dmaiocchi/go/src/github.com/kubic-project/registries-operator/pkg/config/globals.go:21:2: comment on exported var DefaultPrefix should be of the form "DefaultPrefix ..."
/home/dmaiocchi/go/src/github.com/kubic-project/registries-operator/pkg/config/globals.go:24:2: comment on exported var DefaultDeployNumReplicas should be of the form "DefaultDeployNumReplicas ..."
/home/dmaiocchi/go/src/github.com/kubic-project/registries-operator/pkg/test/util.go:1:1: package comment should be of the form "Package test ..."
/home/dmaiocchi/go/src/github.com/kubic-project/registries-operator/pkg/test/util.go:1:1: package comment should not have leading space
/home/dmaiocchi/go/src/github.com/kubic-project/registries-operator/pkg/util/strings.go:26:6: func SafeId should be SafeID
```

--- a/cmd/registries-operator/main.go
+++ b/cmd/registries-operator/main.go
@@ -1,21 +1,19 @@
 /*
- * Copyright 2018 SUSE LINUX GmbH, Nuernberg, Germany..
- *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- *     http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- *
- */
+Copyright 2018 SUSE LINUX GmbH, Nuernberg, Germany..
 
-// Command main is a cli for this k8s operator
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+*/
 package main
 
 import (

--- a/cmd/registries-operator/main.go
+++ b/cmd/registries-operator/main.go
@@ -14,6 +14,8 @@
  * limitations under the License.
  *
  */
+
+// Command main is a cli for this k8s operator
 package main
 
 import (
@@ -39,8 +41,10 @@ import (
 	regcfg "github.com/kubic-project/registries-operator/pkg/config"
 )
 
-// to be set from the build process
+// Version is set from the build process
 var Version string
+
+// Build is set from the build process
 var Build string
 
 // newCmdManager runs the manager

--- a/pkg/apis/kubic/v1beta1/register.go
+++ b/pkg/apis/kubic/v1beta1/register.go
@@ -15,7 +15,6 @@
  *
  */
 
-
 //
 // NOTE: Boilerplate only.  Ignore this file.
 //

--- a/pkg/apis/kubic/v1beta1/registry_types.go
+++ b/pkg/apis/kubic/v1beta1/registry_types.go
@@ -46,7 +46,7 @@ type RegistryStatus struct {
 	Certificate RegistryCertificateStatus
 }
 
-// RegistryStatus defines the observed state of Registry
+// RegistryCertificateStatus defines the observed state of Registry
 type RegistryCertificateStatus struct {
 	// Important: Run "make" to regenerate code after modifying this file
 
@@ -73,7 +73,7 @@ type Registry struct {
 	Status RegistryStatus `json:"status,omitempty"`
 }
 
-// GetRegistryCert gets the certificate for a registry
+// GetCertificateSecret gets the certificate for a registry
 func (registry Registry) GetCertificateSecret(r client.Client) (*v1.Secret, error) {
 	if registry.Spec.Certificate == nil {
 		return nil, nil
@@ -88,8 +88,9 @@ func (registry Registry) GetCertificateSecret(r client.Client) (*v1.Secret, erro
 
 }
 
-func (r Registry) String() string {
-	return fmt.Sprintf("%s", r.Spec.HostPort)
+// String returns Registry HOST:PORT formatted address
+func (registry Registry) String() string {
+	return fmt.Sprintf("%s", registry.Spec.HostPort)
 }
 
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object

--- a/pkg/apis/kubic/v1beta1/registry_types.go
+++ b/pkg/apis/kubic/v1beta1/registry_types.go
@@ -88,7 +88,7 @@ func (registry Registry) GetCertificateSecret(r client.Client) (*v1.Secret, erro
 
 }
 
-// String returns Registry HOST:PORT formatted address
+// String returns registry HOST:PORT formatted address
 func (registry Registry) String() string {
 	return fmt.Sprintf("%s", registry.Spec.HostPort)
 }

--- a/pkg/config/globals.go
+++ b/pkg/config/globals.go
@@ -18,9 +18,9 @@
 package config
 
 var (
-	// A prefix for all the resources created by the operator
+	//DefaultPrefix for all the resources created by the operator
 	DefaultPrefix = "regsop"
 
-	// Default number of replicas for the Deployment
+	// DefaultDeployNumReplicas is the  number of replicas for the Deployment
 	DefaultDeployNumReplicas = 3
 )

--- a/pkg/config/globals.go
+++ b/pkg/config/globals.go
@@ -18,7 +18,7 @@
 package config
 
 var (
-	//DefaultPrefix for all the resources created by the operator
+	// DefaultPrefix for all the resources created by the operator
 	DefaultPrefix = "regsop"
 
 	// DefaultDeployNumReplicas is the  number of replicas for the Deployment

--- a/pkg/controller/registry/registry_cert_installer.go
+++ b/pkg/controller/registry/registry_cert_installer.go
@@ -76,7 +76,7 @@ func (r *ReconcileRegistry) reconcileCertPresent(registry *kubicv1beta1.Registry
 
 	// 3. Process all the Jobs that were launched from this controller
 	jobs, err := getAllJobsWithLabels(r, map[string]string{
-		jobInstallLabelHostPort: kubicutil.SafeId(registry.Spec.HostPort),
+		jobInstallLabelHostPort: kubicutil.SafeID(registry.Spec.HostPort),
 		jobInstallLabelHash:     specSecretHash,
 	})
 	if err != nil {
@@ -164,8 +164,8 @@ func (r *ReconcileRegistry) reconcileCertPresent(registry *kubicv1beta1.Registry
 func (r *ReconcileRegistry) installCertForRegistry(registry *kubicv1beta1.Registry, secret *corev1.Secret, numNodes int) error {
 	var err error
 
-	registryAddress := kubicutil.SafeId(registry.Spec.HostPort)
-	jobName := kubicutil.SafeId(jobInstallNamePrefix) + "-" + registryAddress
+	registryAddress := kubicutil.SafeID(registry.Spec.HostPort)
+	jobName := kubicutil.SafeID(jobInstallNamePrefix) + "-" + registryAddress
 
 	// note: docker cannot mount directories with colons (like "registry.suse.de:5000")
 	//       so we will use the path "/certs/this-registry/ca.crt"

--- a/pkg/controller/registry/registry_cert_remover.go
+++ b/pkg/controller/registry/registry_cert_remover.go
@@ -60,7 +60,7 @@ func (r *ReconcileRegistry) reconcileCertMissing(instance *kubicv1beta1.Registry
 	secretHash := instance.Status.Certificate.CurrentHash
 
 	jobs, err := getAllJobsWithLabels(r, map[string]string{
-		jobRemoveLabelHostPort: kubicutil.SafeId(instance.Spec.HostPort),
+		jobRemoveLabelHostPort: kubicutil.SafeID(instance.Spec.HostPort),
 		jobRemoveLabelHash:     secretHash,
 	})
 	if err != nil {
@@ -123,8 +123,8 @@ func (r *ReconcileRegistry) reconcileCertMissing(instance *kubicv1beta1.Registry
 func (r *ReconcileRegistry) removeCertForRegistry(registry *kubicv1beta1.Registry, secretHash string, numNodes int) error {
 	var err error
 
-	registryAddress := kubicutil.SafeId(registry.Spec.HostPort)
-	jobName := kubicutil.SafeId(jobRemoveNamePrefix) + "-" + registryAddress
+	registryAddress := kubicutil.SafeID(registry.Spec.HostPort)
+	jobName := kubicutil.SafeID(jobRemoveNamePrefix) + "-" + registryAddress
 
 	dstDir := filepath.Join(dockerCertsDir, registry.Spec.HostPort)
 

--- a/pkg/controller/registry/runner.go
+++ b/pkg/controller/registry/runner.go
@@ -126,7 +126,7 @@ func getRunnerJobWithSecrets(cfg *runnerWithSecrets) (*batchv1.Job, error) {
 	jobCont0.VolumeMounts = []corev1.VolumeMount{}
 
 	for hostPort, secret := range cfg.Secrets {
-		name := kubicutil.SafeId(hostPort)
+		name := kubicutil.SafeID(hostPort)
 
 		newVolume := corev1.Volume{
 			Name: name,

--- a/pkg/test/util.go
+++ b/pkg/test/util.go
@@ -15,7 +15,7 @@
  *
  */
 
-// Package test is defined skip the specific test if we are not use running in integration testing mode
+// Package test contains specific test utilities which can be used by operator
 package test
 
 import (

--- a/pkg/test/util.go
+++ b/pkg/test/util.go
@@ -14,6 +14,8 @@
  * limitations under the License.
  *
  */
+
+// Package test is defined skip the specific test if we are not use running in integration testing mode
 package test
 
 import (

--- a/pkg/util/strings.go
+++ b/pkg/util/strings.go
@@ -21,9 +21,9 @@ import (
 	"strings"
 )
 
-// SafeId returns a safe ID (for example, for using in YAML)
+// SafeID returns a safe ID (for example, for using in YAML)
 // ie, "Something:6000/ddd" becomes "something-6000-ddd"
-func SafeId(s string) string {
+func SafeID(s string) string {
 	replacer := strings.NewReplacer(" ", "-", ":", "-", "/", "-", ".", "-")
 	return replacer.Replace(strings.ToLower(s))
 }

